CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,111 @@
+
+<a name="patience"></a>
+
+## [Be patient](#patience)
+
+Sometimes users, admins, and moderators alike will have difficulty in getting across a point they wish to make. This could be for a variety of reasons; Language barrier, technical knowhow, etc. Please be patient with everyone, and remember to [respond to the content of a post, and not the tone](#agreeable).
+
+<a name="cleverclogs"></a>
+
+## [Nobody likes a smart-ass](#cleverclogs)
+
+Spotted something the developers have overlooked? Think you're much cleverer than them for noticing it? Think it's a good idea to post about how clever you are? Top tip: It's not.
+
+Why not open a [pull request](https://github.com/pi-hole/pi-hole/pulls), instead? Pi-hole is opensource software developed by a handful of volunteers with full time jobs, things are bound to get missed! We rely on clear and concise communication from the community at large to help us out when we've missed something, and we all appreciate learning something new! But don't presume you're above anyone because you've had to point something out.
+
+
+<a name="improve"></a>
+
+## [Improve the Discussion](#improve)
+
+Help us make this a great place for discussion by always working to improve the discussion in some way, however small. If you are not sure your post adds to the conversation, think over what you want to say and try again later.
+
+The topics discussed here matter to us, and we want you to act as if they matter to you, too. Be respectful of the topics and the people discussing them, even if you disagree with some of what is being said.
+
+One way to improve the discussion is by discovering ones that are already happening. Please spend some time browsing the topics here before replying or starting your own, and you’ll have a better chance of meeting others who share your interests.
+
+<a name="agreeable"></a>
+
+## [Be Agreeable, Even When You Disagree](#agreeable)
+
+You may wish to respond to something by disagreeing with it. That’s fine. But, remember to _criticize ideas, not people_. Please avoid:
+
+*   Name-calling.
+*   Ad hominem attacks.
+*   Responding to a post’s tone instead of its actual content.
+*   Knee-jerk contradiction.
+
+Instead, provide reasoned counter-arguments that improve the conversation.
+
+<a name="participate"></a>
+
+## [Your Participation Counts](#participate)
+
+The conversations we have here set the tone for everyone. Help us influence the future of this community by choosing to engage in discussions that make this forum an interesting place to be &mdash; and avoiding those that do not.
+
+Discourse provides tools that enable the community to collectively identify the best (and worst) contributions: favorites, bookmarks, likes, flags, replies, edits, and so forth. Use these tools to improve your own experience, and everyone else’s, too.
```

Comment:
Discourse specific

Reply:
Yeah, it needs some modifications and edits.